terminal-emulator/src/main/java/com/termux/terminal/TerminalSession.java

```diff
@@ -115,24 +116,42 @@ public void handleMessage(Message msg) {
                     mEmulator.append(mReceiveBuffer, bytesRead);
                     notifyScreenUpdate();
                 }
-            } else if (msg.what == MSG_PROCESS_EXITED) {
-                int exitCode = (Integer) msg.obj;
-                cleanupResources(exitCode);
-                mChangeCallback.onSessionFinished(TerminalSession.this);
-
-                String exitDescription = "\r\n[Process completed";
-                if (exitCode > 0) {
-                    // Non-zero process exit.
-                    exitDescription += " (code " + exitCode + ")";
-                } else if (exitCode < 0) {
-                    // Negated signal.
-                    exitDescription += " (signal " + (-exitCode) + ")";
-                }
-                exitDescription += " - press Enter]";
-
-                byte[] bytesToWrite = exitDescription.getBytes(StandardCharsets.UTF_8);
-                mEmulator.append(bytesToWrite, bytesToWrite.length);
-                notifyScreenUpdate();
+            }
+            
+            else {
+                if (msg.what == MSG_PROCESS_EXITED) {
+                    File tmpDir = new File("/data/data/com.termux/files/home/.hushlogout");
+                    boolean exists = tmpDir.exists();
+                    if (exists) { 
+                        int exitCode = (Integer) msg.obj;
+                        mShellExitStatus = exitCode;
+                        mShellPid = -1;
+                        mTerminalToProcessIOQueue.close();                         mProcessToTerminalIOQueue.close();
```

Comment:
I'm guessing this shouldn't be here twice?

Reply:
Solved..